apis/v1alpha1/httproute_types.go

```diff
@@ -532,6 +546,45 @@ type HTTPRequestMirrorFilter struct {
 	Port PortNumber `json:"port"`
 }
 
+// HTTPRequestAuthorization defines configuration for the RequestAuthorization filter.
+type HTTPRequestAuthorization struct {
+	// URL indicates the address of the service used to perform authorization checks.
+	//
+	// There are two protocols supported:
+	// - grpc: In this mode, the authorization server should receive a grpc request of type <service.auth.v3.CheckRequest>.
+	//   It is expected for the authorization server to implement the Envoy ExtAuthz GRPC protocol:
```

Comment:
Sorry if I missed some discussion about this, but I am surprised that a generic API is referencing a concrete implementation (envoy). Would we expect, say, an nginx implementor to use the envoy auth?

Reply:
@howardjohn hi!  I'm planning on presenting a design proposal at tomorrow's community meeting, where I will try to explain some of those points to discuss them. 

While this is a concrete implementation, it is an open protocol that can be implemented by "any" authorization server. I agree that the grpc implementation is trickier to support in different proxies and perhaps not possible... But the HTTP version is very "straightforward" to implement in Openresty, and I'm currently working on an implementation for HAProxy.

Reply:
As discussed in the community meeting, it's better to avoid tying this filter to a specific implementation.